docs/docs/user-guide/using-cli-context.md

```diff
@@ -0,0 +1,107 @@
+# Use CLI for Microsoft 365 context
+
+CLI for Microsoft 365 context provides unique functionality to save any kind of option and its value you might use for any command. The options saved in the context may be used in any kind of command execution, that way you may provide less or no options. It may be especially useful in scenarios where you many times use the same option in multiple commands or you would like to group all commands parameters in a single file that are used in the a script file.
+
+## How to get started
+
+In order to create an empty context we may execute the following command:
+```powershell
+m365 context init
+```
+
+To add or update an option in the context use the `m365 context option set` command. The `name` is used to define the option name and `value` is used to define its default value. For example, if we want to set `test list` value for option `listTitle` we should execute:
+```powershell
+m365 context option set --name 'listTitle' --value 'test list'
+```
+
+In order to check what is defined in the context we may use the `m365 context option list` command.
+
+To remove a specific option defined in the context we may run:
+```powershell
+m365 context option remove
+```
+
+In order to remove the full context we may execute the following command:
+```powershell
+m365 context remove
+```
+
+## How does it work
+
+When a command is executed CLI will first check for the `.m365rc.json` file in the current directory (this is where the context is saved once you set any option). If present, CLI will parse the context object and check if any of the options defined in it may be used for the currently executed command. If yes CLI will execute this command with the option and its value taken from the context.
+
+For example if we have the following context defined in the `.m365rc.json` file
+```json
+{
+  "context": {
+    "groupName": "test group",
+    "listTitle": "test list"
+  }
+}
+```
+
+And we will execute the following command:
+```powershell
+m365 spo listitem list --webUrl "https://contoso.sharepoint.com/sites/sample"
+```
+
+The command is missing the `listTitle` required option but instead of failing it will be executed as this option is present in the current context. The `groupName` option will not be used from the context as the `spo listitem list` command does not have it.
+
+When the same option is defined in the context and also in the command itself then the value defined in the command will be used.
+
+## Example
+
+Considering the below context:
```

Comment:
```suggestion
Consider the below context:
```

Reply:
`Consider` is more like you direct the reader to give his attention to what follows. The sentence is complete.
If you use the variant with considering, it looks like the sentence is not complete. You expect to need to read on: "Considering the below context _you can see bla bla bla_" or some such.

Reply:
It's an imperative... the same as:

"Take a look at the following code"